runtime/compiler/runtime/Trampoline.cpp

```diff
@@ -1003,6 +1004,74 @@ void arm64CreateMethodTrampoline(void *trampPtr, void *startPC, void *method)
 #endif
    }
 
+#define ARM64_INSTRUCTION_LENGTH 4
+
+/**
+ * @brief Answers if the callSite is interface call via cache slots
+ *
+ * @param[in]  callSite:             address of the branch instruction at the caller method
+ * @param[out] addrOfFirstClassSlot: address of the first class cache slot
+ * @return true if the callSite is interface call via cache slots
+ */
+static bool isInterfaceCallSite(uint8_t *callSite, intptr_t& addrOfFirstClassSlot)
+   {
+   /*
+    *  Following instruction sequence is used for interface call.
+    *  We can assume tmpReg is x9.
+    *  Searching for the last 4 instructions
+    *
+    *  ldrx tmpReg, L_firstClassCacheSlot
+    *  cmpx vftReg, tmpReg
+    *  ldrx tmpReg, L_firstBranchAddressCacheSlot
+    *  beq  hitLabel
+    *  ldrx tmpReg, L_secondClassCacheSlot
+    *  cmpx vftReg, tmpReg
+    *  bne  snippetLabel
+    *  ldrx tmpReg, L_secondBranchAddressCacheSlot
+    * hitLabel:
+    *  blr  tmpReg
+    * doneLabel:
+    */
+
+   int32_t blrInstr = *reinterpret_cast<int32_t *>(callSite);
+   /* Check if the instruction at the callSite is 'blr x9' */
+   if (blrInstr != 0xd63f0120)
+      {
+      return false;
+      }
+
+   int32_t ldrInst = *reinterpret_cast<int32_t *>(callSite - ARM64_INSTRUCTION_LENGTH);
+   /* Check if the instruction before blr is 'ldr x9, label' */
```

Comment:
The instruction to be checked is `ldrx`.

Reply:
Fixed.